File: 2000/3/7-3.py
#!/usr/bin/env python

import sys

ln = 1
for line in sys.stdin:
    if line == "":
        break
    if line[-1] == "\n":
        line = line[:-1]
    nofreqs = 1
    cnt = 0
    freq = [0] * 26
    for ch in line:
        ch = ch.lower()
        i = ord(ch)
        if ch <= "z" and ch >= "a":
            freq[i - ord("a")] += 1
        cnt += 1
    print "Line %s: " % ln,
    for i in xrange(26):
        if 20 * freq[i] > 3 * cnt:
            print chr(i + ord("A")), "is a super freq.",
            nofreqs = 0
    if nofreqs:
        print "There are no super freqs.",
    print
    ln += 1

File: 2000/3/7-3.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

void main()
{
	ifstream in("prob3_in.txt");
	bool nofreqs;
	int ln = 1, cnt, freq[26] = {0};
	string line;
	char ch;
	while (getline(in, line))
	{
		nofreqs = true;
		cnt = 0;
		for (int i = 0; i < line.length(); i++)
		{
			ch = tolower(line[i]);
			if (isalpha(ch))
				freq[ch - 'a']++;
			cnt++;
		}
		cout << "Line " << ln++ << ": ";
		for (int i = 0; i < 26; i++)
		{
			if ((20 * freq[i]) > (3 * cnt))
			{
				cout << (char)(i + 'A') << " is a super freq. ";
				nofreqs = false;
			}
			freq[i] = 0;
		}
		if (nofreqs)
			cout << "There are no super freqs.";
		cout << endl;
	}
}
